src/ai/backend/agent/docker/kernel.py

```diff
@@ -245,7 +246,14 @@ async def download_file(self, filepath: str):
                     fsize = tarobj.fileobj.tell()
                     if fsize > 1048576:
                         raise ValueError("too large file")
-                    tarbytes = tarobj.fileobj.getvalue()
+                    tarobj.fileobj.seek(0)
+                    tar = tarfile.open(mode="r", fileobj=tarobj.fileobj)
+                    extarobj = tar.extractfile(tar.getnames()[0])
+                    if extarobj:
+                        tarbytes = extarobj.read()
```

Comment:
Again, we don't need to recreate the tarfile object since we already have it in the memory.
```suggestion
                    inner_file = tarobj.extractfile(tarobj.getnames()[0])
                    if inner_file:
                        tarbytes = inner_file.read()
```